chain/xrp/client/client.go

```diff
@@ -0,0 +1,906 @@
+		appendDestinationsForAffectedNodes(txResponse, destinations, "AccountRoot", 6)
+	} else {
+		amount, err := xc.NewAmountHumanReadableFromStr(txResponse.Result.TakerGets.TokenAmount.Value)
+		if err == nil {
+			appendSourceForTakerGetsToken(txResponse, sources, amount)
+			appendDestinationsForAffectedNodes(txResponse, destinations, "RippleState", 15)
+		} else {
+			return
+		}
+	}
+}
+
+// Handle TakerPays logic
+func handleTakerPays(txResponse TransactionResponse, sources, destinations *[]*xc.LegacyTxInfoEndpoint) {
+	if txResponse.Result.TakerPays.XRPAmount != "" {
+		appendSourceForTakerPaysXRP(txResponse, destinations)
+		appendSourcesForAffectedNodes(txResponse, sources, destinations, "AccountRoot", 6)
+	} else {
+		amount, err := xc.NewAmountHumanReadableFromStr(txResponse.Result.TakerPays.TokenAmount.Value)
+		if err == nil {
+			appendDestinationForTakerPaysToken(txResponse, destinations, amount)
+			appendSourcesForAffectedNodes(txResponse, sources, destinations, "RippleState", 15)
+		}
+	}
+}
+
+// Append sources for TakerGets native transactions
+func appendSourceForTakerGetsXRP(txResponse TransactionResponse, sources *[]*xc.LegacyTxInfoEndpoint) {
+	*sources = append(*sources, &xc.LegacyTxInfoEndpoint{
+		Address: xc.Address(txResponse.Result.Account),
+		Amount:  xc.NewAmountBlockchainFromStr(txResponse.Result.TakerGets.XRPAmount),
+	})
+}
+
+// Append destinations for TakerPays native transactions
+func appendSourceForTakerPaysXRP(txResponse TransactionResponse, destinations *[]*xc.LegacyTxInfoEndpoint) {
+	*destinations = append(*destinations, &xc.LegacyTxInfoEndpoint{
+		Address: xc.Address(txResponse.Result.Account),
+		Amount:  xc.NewAmountBlockchainFromStr(txResponse.Result.TakerPays.XRPAmount),
+	})
+}
+
+// Appending sources for TakerGets token transactions
+func appendSourceForTakerGetsToken(txResponse TransactionResponse, sources *[]*xc.LegacyTxInfoEndpoint, amount xc.AmountHumanReadable) {
+	*sources = append(*sources, &xc.LegacyTxInfoEndpoint{
+		Address:         xc.Address(txResponse.Result.TakerGets.TokenAmount.Issuer),
+		ContractAddress: xc.ContractAddress(txResponse.Result.TakerGets.TokenAmount.Issuer + "-" + txResponse.Result.TakerGets.TokenAmount.Currency),
+		Amount:          amount.ToBlockchain(15),
+		Asset:           txResponse.Result.TakerGets.TokenAmount.Currency,
+	})
+}
+
+// Appending destinations for TakerPays token transactions
+func appendDestinationForTakerPaysToken(txResponse TransactionResponse, destinations *[]*xc.LegacyTxInfoEndpoint, amount xc.AmountHumanReadable) {
+	*destinations = append(*destinations, &xc.LegacyTxInfoEndpoint{
+		Address:         xc.Address(txResponse.Result.Account),
+		ContractAddress: xc.ContractAddress(txResponse.Result.TakerPays.TokenAmount.Issuer + "-" + txResponse.Result.TakerPays.TokenAmount.Currency),
+		Amount:          amount.ToBlockchain(15),
+		Asset:           txResponse.Result.TakerPays.TokenAmount.Currency,
+	})
+}
```

Comment:
I the function is only appending to a list, it's kind of too simple and makes it harder to read.  I think It'd be easier to read to just leave these inlined.